### PR TITLE
ENH Hide the open in new window checkbox from phone and email links

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/23

### CODE REVIEW

File: src/Controllers/LinkFieldController.php [hunk 1]
```diff
@@ -356,8 +356,8 @@ private function createLinkForm(Link $link, string $operation): Form
 
         // Add save action button
         $title = $id
-                ? _t(__CLASS__ . '.UPDATE_LINK', 'Update link')
-                : _t(__CLASS__ . '.CREATE_LINK', 'Create link');
+            ? _t(__CLASS__ . '.UPDATE_LINK', 'Update link')
+            : _t(__CLASS__ . '.CREATE_LINK', 'Create link');
```
Comment (emteknetnz) [inline src/Controllers/LinkFieldController.php:360]: This is unrelated refactoring

File: tests/php/Traits/AllowedLinkClassesTraitTest.php [hunk 2]
```diff
@@ -25,12 +25,12 @@ class AllowedLinkClassesTraitTest extends SapphireTest
...
-        FileLink::class,
-        EmailLink::class,
-        PhoneLink::class,
-        TestPhoneLink::class,
+        'sitetree' => SiteTreeLink::class,
+        'external' => ExternalLink::class,
+        'file' => FileLink::class,
+        'email' => EmailLink::class,
+        'phone' => PhoneLink::class,
+        'testphone' => TestPhoneLink::class,
```
Comment (emteknetnz) [inline tests/php/Traits/AllowedLinkClassesTraitTest.php:33]: This is unrelated refactoring to make it easier to run unit tests locally when you have a custom link type in your project

File: tests/php/Traits/AllowedLinkClassesTraitTest.php [hunk 3]
```diff
@@ -58,20 +58,20 @@ public function allowedTypesDataProvider() : array
...
-                  FileLink::class,
-                  EmailLink::class,
-                  PhoneLink::class,
-                  TestPhoneLink::class,
+                    SiteTreeLink::class,
+                    ExternalLink::class,
+                    FileLink::class,
+                    EmailLink::class,
+                    PhoneLink::class,
+                    TestPhoneLink::class,
```
Comment (emteknetnz) [inline tests/php/Traits/AllowedLinkClassesTraitTest.php:66]: This is unrelated refactoring

File: tests/php/Traits/AllowedLinkClassesTraitTest.php [hunk 4]
```diff
@@ -58,20 +58,20 @@ public function allowedTypesDataProvider() : array
...
```
Comment (emteknetnz) [inline tests/php/Traits/AllowedLinkClassesTraitTest.php:74]: This is unrelated refactoring

File: tests/php/Traits/AllowedLinkClassesTraitTest.php [hunk 5]
```diff
@@ -174,10 +174,11 @@ public function testGetSortedTypeProps(array $enabled, array $expected, bool $re
         if ($reorder) {
             Injector::inst()->get(TestPhoneLink::class)->config()->set('menu_priority', 5);
         }
-        
+
         $linkField = LinkField::create('LinkField');
         $linkField->setAllowedTypes($enabled);
         $json = json_decode($linkField->getTypesProps(), true);
+        $json = $this->removeCustomLinkTypes($json);
```
Comment (emteknetnz) [inline tests/php/Traits/AllowedLinkClassesTraitTest.php:181]: This is unrelated refactoring to make it easier to run unit tests locally when you have a custom link type in your project

File: src/Models/Link.php [hunk 6]
```diff
@@ -123,6 +126,15 @@ public function getCMSFields(): FieldList
...
             }
         });
+        $this->afterUpdateCMSFields(function (FieldList $fields) {
+            // Move the OpenInNew field to the bottom of the form and update its title
+            $openInNewField = $fields->dataFieldByName('OpenInNew');
+            $fields->removeByName('OpenInNew');
+            if ($this->showOpenInNewWindowField) {
+                $fields->addFieldToTab('Root.Main', $openInNewField);
+                $openInNewField->setTitle(_t(__CLASS__ . '.OPEN_IN_NEW_TITLE', 'Open in new window?'));
+            }
```
Comment (ChloeSartorelli) [inline src/Models/Link.php]: This will throw away title changes people have made in their custom link types or extensions.

Instead of this, keep the old lines 106 and 107, and change this to:
```suggestion
            // Move the OpenInNew field to the bottom of the form and update its title
            $openInNewField = $fields->dataFieldByName('OpenInNew');
            if ($openInNewField) {
                $fields->removeByName('OpenInNew');
                $fields->addFieldToTab('Root.Main', $openInNewField);
            }
```
So:
1. we always set the title by default
2. if developers don't want it, they can just remove it (no need for that protected property)
3. only remove it if it's there
4. don't change the title, because we're keeping line 107 (and also keeping any changes anyone makes in extensions or subclasses)

File: src/Models/Link.php [hunk 7]
```diff
@@ -70,6 +70,11 @@ class Link extends DataObject
      */
     private static $icon = 'font-icon-link';
 
+    /**
+     * Whether or not to show the "Open in new window?" checkbox field in the link edit modal
+     */
+    protected bool $showOpenInNewWindowField = true;
```
Comment (ChloeSartorelli) [inline src/Models/Link.php]:
```suggestion
```
Not needed - people can just remove the field if they don't want it, and keep it if they wanna keep it.
Please also remove this from the other links, and add a `$fields->removeByName('OpenInNew');` line to the `getCMSFields()` methods of those link types instead.

Comment (ChloeSartorelli) [reply src/Models/Link.php]: Please also remove the docs that relate to this.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Models/Link.php (hunk 6 above)
Comment (emteknetnz) [reply src/Models/Link.php]: Updated

File: src/Models/Link.php (hunk 7 above)
Comment (emteknetnz) [reply src/Models/Link.php]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM
